Add Container Gateway info (#443)
theforeman/foreman-documentation: guides/doc-Content_Management_Guide/topics/Containers_interacting_with_registry.adoc

[[using_container_registries]]
= Using Katello Container Registries

Podman and Docker can be used to fetch content from Katello container registries.

.Container Registries on Smart Proxies

> On Smart Proxies with content, the https://github.com/Katello/smart_proxy_container_gateway[Container Gateway] smart proxy plugin acts as the container registry.  It caches authentication information from Katello and proxies incoming requests to Pulp.  The Container Gateway comes default on Smart Proxies with content.

Reviewer: ```suggestion
On {SmartProxies} with content, the https://github.com/Katello/smart_proxy_container_gateway[Container Gateway] {SmartProxy} plugin acts as the container registry.  It caches authentication information from Katello and proxies incoming requests to Pulp.  The Container Gateway comes default on {SmartProxies} with content.
```

On Smart Proxies with content, the https://github.com/Katello/smart_proxy_container_gateway[Container Gateway] smart proxy plugin acts as the container registry.  It caches authentication information from Katello and proxies incoming requests to Pulp.  The Container Gateway comes default on Smart Proxies with content.

> *Please note* that, at first release with Katello 4.0, the Container Gateway will not yet support pulling container images that require authentication.  Until it does, please ensure that *Unauthenticated Pull* is checked for all Lifecycle Enviroments that contain container repositories that are expected to be served by Smart Proxies.  This is planned to be resolved by Katello 4.1.

Reviewer 2: I think we should introduce Limitations header like we often do in other cases. That's usually better way to document limitations.

.Procedure

Logging in to the container registry:

----
podman login `hostname`/
----

Listing container images:

----

> podman search `hostname`/

Reviewer 2: Please use "example.com", we have couple of macros we use in snippets like "foreman.example.com" or "proxy.example.com" available. Also is the trailing slash mandatory?

Author: It wasn't needed for `login` but it is needed to list all images on a certain registry. I've updated the slashes.

----
podman pull {foreman-example-com}/my-image:<optional_tag>
----

.Limitations

> Please note that, at first release with Katello 4.0, the Container Gateway will not yet support pulling container images that require authentication.  Until it does, ensure that *Unauthenticated Pull* is checked for all Lifecycle Enviroments that have container repositories that are expected to be served by Smart Proxies.  This is planned to be resolved by Katello 4.1.

Reviewer: ```suggestion
With the Katello 4.0 release, the Container Gateway does not support pulling container images that require authentication.  Until it does, ensure that *Unauthenticated Pull* is checked for all Lifecycle Environments that have container repositories that are expected to be served by {SmartProxies}.  This is planned to be resolved by Katello 4.1.
```

I would add the link to the redmine issue and say "Until x is resolved " rather than make any promises. Can you please add an ifdef for katello around this bit for fear it would get into Satellite docs in time :)
